### PR TITLE
Add path and pattern options to views

### PR DESCRIPTION
**Pull request by chrismccord**
Allows wildcard template inclusion as well as
customized template directory locations

### CODE REVIEW

**Comment by chrismccord** [comment]
@josevalim signoff please <3

File: lib/phoenix/view.ex [hunk 1]
```diff
@@ -95,54 +95,46 @@ defmodule Phoenix.View do
   can be further configured in the Phoenix application. You can read
   more about format encoders in `Phoenix.Template` documentation.
   """
+  alias Phoenix.{Template}
```
Comment (josevalim) [inline lib/phoenix/view.ex]: ? :)

**Comment by josevalim** [comment]
:shipit:

File: lib/phoenix/view.ex (hunk 1 above)
Comment (chrismccord) [inline lib/phoenix/view.ex]: I *knew* you would not like this, but I had an overly long line and I had to get rid of it! Don't judge me

Comment (jeregrine) [inline lib/phoenix/view.ex]: You could do `alias Phoenix.Template` without the `{}`

File: lib/phoenix/view.ex [hunk 2]
```diff
@@ -95,54 +95,46 @@ defmodule Phoenix.View do
...
 
   @doc """
   When used, defines the current module as a main view module.
 
   ## Options
 
     * `:root` - the template root to find templates
+    * `:path` - the optional path to search for templates within the `:root`.
+      Defaults to the underscored view module name. A blank string may
+      be provided to use the `:root` path direclty as the template lookup path.
```
Comment (Gazler) [inline lib/phoenix/view.ex]: direclty -> directly

File: lib/phoenix/view.ex [hunk 3]
```diff
@@ -95,54 +95,46 @@ defmodule Phoenix.View do
...
   The `:root` option is required while the `:namespace` defaults to the
   first nesting in the module name. For instance, both `MyApp.UserView`
   and `MyApp.Admin.UserView` have namespace `MyApp`.
 
-  The namespace is used to calculate paths. For example, if you are in
-  `MyApp.UserView` and the namespace is `MyApp`, templates are expected
-  at `Path.join(root, "user")`. On the other hand, if the view is
-  `MyApp.Admin.UserView`, the path will be `Path.join(root, "admin/user")`
-  and so on.
+  The `:namespace` and `:path` options are is used to calculate template
```
Comment (Gazler) [inline lib/phoenix/view.ex]: are is  -> are

**Pull request merged**